src/components/ui/redux-paginstion-slice/fetch-slice.ts

```diff
@@ -0,0 +1,34 @@
+import { createApi, fetchBaseQuery } from "@reduxjs/toolkit/query/react";
```

Comment:
it's not correct to keep store-related files in the `components/ui` folder

please move into the separate to `store` or `redux` folder